### PR TITLE
Args everywhere: finetune scripts

### PR DESCRIPTION
**Pull request by carmocca**

### CODE REVIEW

File: finetune/full.py [hunk 1]
```diff
@@ -74,47 +77,72 @@ def setup(
...
-    state = {
-        "model": model,
-        "optimizer": optimizer,
-        "scheduler": scheduler,
-        "hparams": hparams,
-        "iter_num": 0,
-        "step_count": 0,
-    }
+    scheduler = get_lr_scheduler(optimizer, warmup_steps=lr_warmup_steps, max_steps=lr_max_steps)
+    state = {"model": model, "optimizer": optimizer, "scheduler": scheduler, "iter_num": 0, "step_count": 0}
```
Comment (carmocca) [inline finetune/full.py:142]: @awaelchli Do you want this state to be made into an actual class or should we keep it freeform for now?

Comment (awaelchli) [reply finetune/full.py:142]: I would prefer freeform or AttributeDict. Fabric needs it in a dict anyway, and these aren't configurable by the CLI.

File: finetune/lora.py [hunk 2]
```diff
@@ -95,36 +89,64 @@ def setup(
...
+            dropout=lora_dropout,
+            to_query=lora_query,
+            to_key=lora_key,
+            to_value=lora_value,
+            to_projection=lora_projection,
+            to_mlp=lora_mlp,
+            to_head=lora_head,
+        ),
+        IOArgs(data_dir, checkpoint_dir, out_dir),
+        TrainArgs(
```
Comment (awaelchli) [inline finetune/lora.py:111]: Is this granularity for practical reasons? Would we consider merging train args, eval args and optimization args? They are all train args IMO. In other words, when would I want to only use optimizer args?

Comment (lantiga) [reply finetune/lora.py:111]: I agree, let’s aggregate where we can. Less cognitive overhead.

Comment (carmocca) [reply finetune/lora.py:111]: For instance, in a sweep over lr you'd only need to change one small config

Ill merge this as is and when all the scripts have been ported, revisit this choice.

**Pull request merged**